### PR TITLE
Issue 184: Support GitHub issues in bk-merge script

### PR DESCRIPTION
**Pull request by sijie**
- Recoganize the issue title and normalize it
- Close the corresponding issue when closing the pr

### CODE REVIEW

**Comment by sijie** [comment]
The script change for #184 . @eolivelli

**Comment by eolivelli** [comment]
+1 I tested it and it is working fine.
Awesome !

File: dev/bk-merge-pr.py [hunk 1]
```diff
@@ -211,6 +212,13 @@ def merge_pr(pr_num, target_ref, title, body, default_pr_reviewers, pr_repo_desc
 
     # The string "Closes #%s" string is required for GitHub to correctly close the PR
     close_line = "Closes #%s from %s" % (pr_num, pr_repo_desc)
+    # Find the github issues to close
+    github_issues = re.findall("%s #[0-9]{3,6}" % GITHUB_ISSUES_NAME, title)
+
+    if len(github_issues) != 0:
+        for issue_id in github_issues:
+            close_line += " and Closes #%s" % (issue_id)
```
Comment (eolivelli) [inline dev/bk-merge-pr.py]: last minute change request: it appears to me that a '#' character is the in the end of the message: "and Closes #ISSUE #191'"

like
`['git', 'commit', '--author="jiazhai <zhaijia@live.com>"', '-m', u'ISSUE #191: add template for ISSUE and PR', '-m', u'Following BP-9, provide a template for ISSUE and PR.\r\nWhen create issue, will use ISSUE template; while create PR will use PR template.', '-m', 'Author: jiazhai <zhaijia@live.com>', '-m', 'Reviewers: Enrico Olivelli <None>', '-m', u'Closes #192 from jiazhai/issue_191 and Closes #ISSUE #191']
`

maybe is is better to drop the '#' before in "#ISSUE#

I was trying to do a dry run on merging #192 from Jia

Comment (sijie) [inline dev/bk-merge-pr.py]: oh. my bad. it should be "closes #<issue-id>".

Comment (sijie) [inline dev/bk-merge-pr.py]: done, fixed this.

"This closes #<pr-num> from <pr-branch-name>, closes #<issue-id> and ..."

**Comment by jiazhai** [comment]
cool

**Pull request closed without merging**